src/background/target-page-controller.ts

```diff
@@ -142,29 +144,34 @@ export class TargetPageController {
 
     private handleTabUrlUpdate = async (tabId: number): Promise<void> => {
         this.tabContextManager.addTabContextIfNotExists(tabId, this.tabContextFactory);
-        this.sendTabUrlUpdatedAction(tabId);
+        await this.sendTabUrlUpdatedAction(tabId);
         await this.addKnownTabId(tabId);
     };
 
-    private sendTabUrlUpdatedAction(tabId: number): void {
-        this.browserAdapter.getTab(
-            tabId,
-            (tab: chrome.tabs.Tab) => {
-                this.tabContextManager.interpretMessageForTab(tabId, {
-                    messageType: Messages.Tab.ExistingTabUpdated,
-                    payload: tab,
-                    tabId: tabId,
-                });
-            },
-            () => {
-                this.logger.log(
-                    `sendTabUrlUpdatedAction: tab with ID ${tabId} not found, skipping action message`,
-                );
-            },
-        );
+    private async sendTabUrlUpdatedAction(tabId: number): Promise<void> {
+        await new Promise<void>((resolve, reject) => {
```

Comment:
With this change, half the callsites for `BrowserAdapter.getTab` are now doing this promise-wrapping dance - I wonder if it would be more readable/concise to instead switch `BroserAdapter.getTab` to be promise-based

Reply:
I added a getTabAsync function to BrowserAdapter to wrap this call in a promise, but currently there's one getTab use in AssessmentStore that's complicated to replace. Switching over to use a promise there would make some of the action listeners async and complicate the testing code quite a bit. It will be much easier to replace once the planned changes to stores and actions have been made.